### PR TITLE
(BKR-1385) Install rpm packages on aix

### PR DESCRIPTION
**Pull request by melissa**

### CODE REVIEW

**Comment by melissa** [comment]
This code installs puppet as expected

```
  Platforms available for this build:
  ["aix-6.1-power", "aix-7.1-power", "cisco-wrlinux-5-x86_64", "cisco-wrlinux-7-x86_64", "cumulus-2.2-amd64", "debian-7-i386", "debian-7-amd64", "debian-8-i386", "debian-8-amd64", "debian-9-i386", "debian-9-
amd64", "el-5-i386", "el-5-x86_64", "el-6-i386", "el-6-x86_64", "el-6-s390x", "el-7-x86_64", "el-7-s390x", "el-7-ppc64le", "el-7-aarch64", "eos-4-i386", "fedora-25-i386", "fedora-25-x86_64", "fedora-26-x86_6
4", "osx-10.10-x86_64", "osx-10.11-x86_64", "osx-10.12-x86_64", "sles-11-i386", "sles-11-x86_64", "sles-11-s390x", "sles-12-x86_64", "sles-12-s390x", "sles-12-ppc64le", "solaris-10-i386", "solaris-10-sparc",
 "solaris-11-i386", "solaris-11-sparc", "ubuntu-14.04-i386", "ubuntu-14.04-amd64", "ubuntu-16.04-i386", "ubuntu-16.04-amd64", "ubuntu-16.04-ppc64el", "windows-2012-x86", "windows-2012-x64"]
  PLATFORM SPECIFIC INFO for aix71-8.delivery.puppetlabs.net (packaging name 'aix-7.1-power'):
  - {:artifact=>"./aix/7.1/PC1/ppc/puppet-agent-1.10.8.145.gcf0338d-1.aix7.1.ppc.rpm", :repo_config=>nil}, isnil? false
  - artifact url: 'http://builds.delivery.puppetlabs.net/puppet-agent/cf0338d9bd0c79d76d8c0405d29a2acf5e497b31/artifacts/./aix/7.1/PC1/ppc/puppet-agent-1.10.8.145.gcf0338d-1.aix7.1.ppc.rpm'. Exists? true
  Fetching: http://builds.delivery.puppetlabs.net/puppet-agent/cf0338d9bd0c79d76d8c0405d29a2acf5e497b31/artifacts/./aix/7.1/PC1/ppc/puppet-agent-1.10.8.145.gcf0338d-1.aix7.1.ppc.rpm
    and saving to ./puppet-agent-1.10.8.145.gcf0338d-1.aix7.1.ppc.rpm

  aix71-8.delivery.puppetlabs.net 13:40:53$ rndnum=${RANDOM} && mkdir /tmp/puppet_installer.${rndnum} && echo /tmp/puppet_installer.${rndnum}
    Warning: Skipping ip method to ssh to host as its value is not set. Refer to https://github.com/puppetlabs/beaker/tree/master/docs/how_to/ssh_connection_preference.md to remove this warning
    Warning: Skipping vmhostname method to ssh to host as its value is not set. Refer to https://github.com/puppetlabs/beaker/tree/master/docs/how_to/ssh_connection_preference.md to remove this warning
    Attempting ssh connection to aix71-8.delivery.puppetlabs.net, user: root, opts: {:config=>false, :paranoid=>false, :auth_methods=>["publickey"], :port=>22, :forward_agent=>true, :keys=>["id_rsa_acceptanc
e", "/Users/melissa/.ssh/id_rsa-acceptance"], :user_known_hosts_file=>"/Users/melissa/.ssh/known_hosts", :keepalive=>true, :logger=>#<Logger:0x007fdcb0204880 @progname=nil, @level=4, @default_formatter=#<Log
ger::Formatter:0x007fdcb0204f60 @datetime_format=nil>, @formatter=nil, @logdev=#<Logger::LogDevice:0x007fdcb02057f8 @shift_size=nil, @shift_age=nil, @filename=nil, @dev=#<IO:<STDERR>>, @mutex=#<Logger::LogDe
vice::LogDeviceMutex:0x007fdcb0205a50 @mon_owner=nil, @mon_count=0, @mon_mutex=#<Mutex:0x007fdcb0205f00>>>>, :password_prompt=>#<Net::SSH::Prompt:0x007fdcb0206608>, :user=>"root"}
:paranoid is deprecated, please use :verify_host_key. Supported values are exactly the same, only the name of the option has changed.
    /tmp/puppet_installer.29659

  aix71-8.delivery.puppetlabs.net executed in 0.41 seconds
  localhost $ scp puppet-agent-1.10.8.145.gcf0338d-1.aix7.1.ppc.rpm aix71-8.delivery.puppetlabs.net:/tmp/puppet_installer.29659 {:ignore => }

  aix71-8.delivery.puppetlabs.net 13:42:28$ rpm -ivh  /tmp/puppet_installer.29659/puppet-agent-1.10.8.145.gcf0338d-1.aix7.1.ppc.rpm
    puppet-agent                    #    #    #    #    #    #    #    #    #    #    #    #    #    #    #    #    #    #    #    #    #    #    #    #    #    #    #    #    #    #    #    #    #    #    #
    #    #    #    #    #    #    #    #    #    #    #    #    #    #    #

  aix71-8.delivery.puppetlabs.net executed in 11.08 seconds
```


But something is missing in terms of getting the environment set up correctly
```
aix71-8.delivery.puppetlabs.net 13:43:05$ puppet --version
  bash: puppet: command not found

aix71-8.delivery.puppetlabs.net executed in 0.03 seconds
Exited: 127
Beaker::Host::CommandFailure: Host 'aix71-8.delivery.puppetlabs.net' exited with 127 running:
 puppet --version
Last 10 lines of output were:
        bash: puppet: command not found
/Users/Melissa/beaker/lib/beaker/host.rb:375
/Users/Melissa/beaker/lib/beaker/dsl/helpers/host_helpers.rb:83
/Users/Melissa/beaker/lib/beaker/shared/host_manager.rb:127
/Users/Melissa/beaker/lib/beaker/dsl/patterns.rb:37
/Users/Melissa/beaker/lib/beaker/dsl/helpers/host_helpers.rb:63
/Users/Melissa/puppet/acceptance/setup/aio/pre-suite/010_Install.rb:79
/Users/Melissa/puppet/acceptance/setup/aio/pre-suite/010_Install.rb:78
/Users/Melissa/puppet/acceptance/setup/aio/pre-suite/010_Install.rb:78
/Users/Melissa/beaker/lib/beaker/test_case.rb:133
/Users/Melissa/beaker/lib/beaker/test_case.rb:133
```

File: spec/beaker-puppet/install_utils/puppet5_spec.rb [hunk 1]
```diff
@@ -199,6 +200,19 @@ def run_shared_test_steps
         run_shared_test_steps()
       end
 
+      it 'AIX: fetches the file & runs local install' do
+        @platform = 'aix'
+        @version = '7.2'
```
Comment (joshcooper) [inline spec/beaker-puppet/install_utils/puppet5_spec.rb]: Could you add a separate test for 7.1, as we only do the `ignoreos` part on 7.2.

File: spec/beaker-puppet/install_utils/puppet5_spec.rb [hunk 2]
```diff
@@ -199,6 +200,33 @@ def run_shared_test_steps
...
+      end
+
+      it 'AIX 6.1: fetches the file & runs local install' do
+        @platform = 'aix'
+        @version = '6.1'
+
+        expect( subject ).to receive( :fetch_http_file ).once
+        expect( subject ).to receive( :scp_to ).once
+        expect( subject ).to receive( :on ).with( host, /^rpm -ivh .*#{File.basename(artifact_url)}$/ ).once
+        expect( subject ).not_to receive( :on ).with( host, /^rpm -ivh --ignoreos .*#{File.basename(artifact_url)}$/ )
```
Comment (melissa) [inline spec/beaker-puppet/install_utils/puppet5_spec.rb]: @joshcooper done!

**Comment by joshcooper** [comment]
@tvpartytonight or @kevpl can we haz merge?

**Comment by melissa** [comment]
p.s., the failure I was seeing @joshcooper couldn't replicate... so maybe it's something about my environment that caused that

**Comment by mchllweeks** [comment]
@puppetlabs-jenkins test this please

**Comment by joshcooper** [comment]
ping @mchllweeks

**Comment by mchllweeks** [comment]
@puppetlabs-jenkins test this please

**Comment by mchllweeks** [comment]
I am now on the white list to kick off tests.  Will try again.

**Comment by mchllweeks** [comment]
@puppetlabs-jenkins test this please

**Pull request merged**